patches/kexec-2.0.12.patch

```diff
@@ -0,0 +1,67 @@
+diff -u --recursive clean/kexec-tools-2.0.12/kexec/arch/i386/kexec-x86-common.c build/kexec-tools-2.0.12/kexec/arch/i386/kexec-x86-common.c
+--- clean/kexec-tools-2.0.12/kexec/arch/i386/kexec-x86-common.c	2015-11-29 18:52:07.000000000 -0500
++++ kexec-tools-2.0.12/kexec/arch/i386/kexec-x86-common.c	2017-01-04 13:46:53.684313636 -0500
+@@ -24,6 +24,7 @@
+ #include <fcntl.h>
+ #include <stddef.h>
+ #include <stdio.h>
++#include <inttypes.h>
+ #include <errno.h>
+ #include <stdint.h>
+ #include <string.h>
+@@ -74,20 +75,20 @@
+ 		return -1;
+ 	}
+ 	while(fgets(line, sizeof(line), fp) != 0) {
+-		unsigned long long start, end;
++		uintptr_t start, end;
+ 		char *str;
+ 		int type;
+ 		int consumed;
+ 		int count;
+ 		if (memory_ranges >= MAX_MEMORY_RANGES)
+ 			break;
+-		count = sscanf(line, "%Lx-%Lx : %n",
++		count = sscanf(line, "%"SCNxPTR"-%"SCNxPTR" : %n",
```

Comment:
Strictly speaking it's a conformance bug in applications; on the glibc side it's just doing something unhelpful in the case of UB and masking the bug by guessing what the programmer wanted and doing it rather than reporting the error. The fix looks correct though.